src/include/ssh.h

```diff
@@ -44,45 +44,80 @@
 #ifndef SSH_H
 #define SSH_H
 
-#include <stdio.h>   /* for FILE* */
+#include <stdio.h>      /* for FILE* */
 #include <pcap.h>
 #include "output.h"
 #include "feature.h"
-#include "utils.h"
+#include "utils.h"      /* for enum role */
 
 #define ssh_usage "  ssh=1                      report ssh information\n"
 
 #define ssh_filter(key) ((key->prot == 6) && (key->dp == 22 || key->sp == 22))
 
-#define MAX_SSH_LEN 512
+#define MAX_SSH_STRING_LEN 512
+#define MAX_SSH_KEX_MESSAGES 2 /* large enough for DH, RSA, and GEX key exchanges */
+#define MAX_SSH_PACKET_LEN 35000 /* RFC 4253, Section 6.1. */
+#define MAX_SSH_PAYLOAD_LEN 32768 /* RFC 4253, Section 6.1. */
+
+struct vector {
+    unsigned int len;
+    void *bytes;
+};
+
+void copy_printable_string(char *buf, unsigned buflen, const void *data, unsigned datalen);
```

Comment:
Thanks for moving these functions. Could you make them static (private) within the ssh.c file instead of publicly visible here?